### ISSUE TITLE
Missing lookup function for Values in Advanced Filter

### ISSUE THREAD

**Issue by martin-networker** [nautobot/nautobot#3029]
### Environment
* Nautobot version (Docker tag too if applicable): 1.5.2
* Python version: 3.10
* Database platform, version: -
* Middleware(s): -

### Steps to Reproduce
1. Click the "Filter" button for example in Rack Elevation (it affects also Device and others)
2. Click the "Advanced" tab
3. Select for example name in (same issue also for custom field in Device) 
4. Select contain in Lookup type
5. Value does not offer anything, you have to open a new tab and find values from there.

### Expected Behavior
Value field will offer possible values.

### Observed Behavior
For some filters the lookup function does not work.

**Comment by bryanculver** [comment]
Contains is a substring search. By nature it won't populate exact names for things as it would be perceived that you would be looking for the name containing a string, such as "105":


https://user-images.githubusercontent.com/31187/208682003-dbb5928d-c5fa-4483-b0ff-020a6c4fd418.mov

**Comment by bryanculver** [comment]
However "name" with "exact" lookup expression could be improved to return the list of device names to provide as options.

Many of the advanced filters take advantage of user-supplied strings for searches that previously weren't possible for the default filters before. As an example, the contains and regex lookup expressions weren't possible to put in a form because you would have only been able to search on one provided string. Multiple values entered into the "value" column are `OR` conditions.

**Comment by martin-networker** [comment]
You are right that for `contains` and `regex` the lookup can't work. Sorry for bad example.
However for `exact` it should work and we will be glad if this will be available.
